### PR TITLE
unit: remove explicitly prefixed constants

### LINKED ISSUE #90: unit: clean up unit package
The unit package has a lot of problems. These should be cleaned up. Here are my suggestions

- [x] Clean up basic documentation so that it actually matches the code
- [x] Remove the (incorrect) comment examples and make them test Examples
- [x] Do not provide so many constants for the units. Not `unit.Millimeter`, instead just supply a `unit.Milli` constant, so it can be `unit.Milli * unit.Meter`. Slightly less pretty, but drastically reduces the API surface and makes it more composable with outside code
- [ ] ~~Redefine the `Unit` combination methods to match the `mat` types. For example, `Add` should take in two arguments and store to the receiver~~
- [x] Remove the `formatted` field from `Unit`. It introduces unnecessary state
- [x] Update the autogeneration to use `go generate`. 
- [x] It seems like the formatting code for unit types needs fixing.

### PR DESCRIPTION
**Pull request by kortschak**
Please take a look.

Closes #90.

### CODE REVIEW

File: unit/generate_unit.go [hunk 1]
```diff
@@ -18,16 +18,15 @@ import (
 )
 
 type Unit struct {
-	Name          string
+	DimensionName string
 	Receiver      string
-	Offset        int    // From normal (for example, mass base unit is kg, not kg)
+	PowerOffset   int    // From normal (for example, mass base unit is kg, not kg)
```
Comment (vladimir-ch) [inline unit/generate_unit.go]: is kg, not kg?

File: unit/generate_unit.go [hunk 2]
```diff
@@ -262,54 +169,51 @@ var Units = []Unit{
...
 	},
 	{
-		Name:        "Area",
-		Receiver:    "a",
-		PrintString: "m^2",
-		TypeComment: "Area represents and area in square metres",
+		DimensionName: "Area",
+		Receiver:      "a",
+		PrintString:   "m^2",
+		TypeComment:   "Area represents and area in square metres",
```
Comment (vladimir-ch) [inline unit/generate_unit.go]: an area

File: unit/generate_unit.go (hunk 1 above)
Comment (kortschak) [reply unit/generate_unit.go]: Ha!

Fixed

File: unit/generate_unit.go (hunk 2 above)
Comment (kortschak) [reply unit/generate_unit.go]: Fixed

**Comment by kortschak** [comment]
PTAL

**Comment by kortschak** [comment]
I'll wait.

**Comment by btracey** [comment]
That's such an obvious solution to the Kg problem ... can't believe I didn't think of that.

**Comment by kortschak** [comment]
Which part, the extra constant or the 1e-3?

**Comment by btracey** [comment]
Setting the gram to be 1e-3.

**Comment by kortschak** [comment]
When you remove all the constants, that's what was left. It seemed obvious.

**Comment by btracey** [comment]
I agree, it does seem obvious

**Pull request merged**